src/messages/mod.rs

```diff
@@ -71,39 +93,53 @@ pub struct HopMessageWithBytes {
 
 impl HopMessageWithBytes {
     /// Serialize message and keep both SignedRoutingMessage and Bytes.
-    pub fn new(content: SignedRoutingMessage) -> Result<Self> {
-        let message = Message::Hop(content);
+    pub fn new(full_content: SignedRoutingMessage, log_ident: &LogIdent) -> Result<Self> {
+        let message = Message::Hop(full_content);
         let full_message_bytes = to_network_bytes(&message)?;
 
-        let content = if let Message::Hop(content) = message {
-            content
+        let full_content = if let Message::Hop(full_content) = message {
+            full_content
         } else {
             unreachable!("Created as Hop can only match Hop.")
         };
 
-        Ok(Self::new_from_parts(content, full_message_bytes))
-    }
-
-    fn new_from_parts(content: SignedRoutingMessage, full_message_bytes: Bytes) -> Self {
+        let peek_msg = PeekedSignedRoutingMessageInfo {
+            dst: full_content.routing_message().dst,
+        };
+        let message = Self::new_from_parts(Some(full_content), peek_msg, full_message_bytes);
```

Comment:
the above message is `Message::Hop(full_content)` , and here is `new_from_parts` 
if they are different, will it be better to have a different name ?